models/dimension.go

```diff
@@ -0,0 +1,48 @@
+package models
+
+import (
+	"errors"
+)
+
+// Dimension struct encapsulating Dimension details.
+type Dimension struct {
+	DimensionID string
+	Option      string
+	NodeID      string
+}
+
+// Validate checks the dimension object
+func (d *Dimension) Validate() error {
+	if d == nil {
+		return errors.New("dimension is required but was nil")
+	}
+	if len(d.DimensionID) == 0 && len(d.Option) == 0 {
+		return errors.New("dimension invalid: both dimension.dimension_id and dimension.value are required but were both empty")
+	}
+	if len(d.DimensionID) == 0 {
+		return errors.New("dimension id is required but was empty")
+	}
+	if len(d.Option) == 0 {
+		return errors.New("dimension value is required but was empty")
+	}
+	return nil
+}
+
+// Instance struct to hold instance information.
+type Instance struct {
+	InstanceID string
+	CSVHeader  []string
+	Dimensions []interface{}
```

Comment:
i worry about this being an interface rather than a defined type, is this to allow for dimensions to be different shapes in different contexts, or do we not have a model for the data that will be returned here?

Reply:
I didn't have a model to begin with, it was defined as an interface; I could do some digging to better describe what Dimensions will look like instead of using an interface?

Reply:
nah if it wasnt defined before its fine 👍